src/pages/ReactionPage/ReactionPage.tsx

```diff
@@ -17,9 +18,8 @@ const ReactionPage = () => {
   const partygoers = useSelector(partygoersSelector);
   const usersById = partygoers;
   const reactions = useSelector((state) => state.firestore.ordered.reactions);
-  const chats = useSelector((state) =>
-    state.firestore.ordered.venueChats?.filter((chat) => chat.deleted !== true)
-  );
+  const chats = useVenueChats();
+  const filteredChats = chats?.filter((chat) => chat.deleted !== true);
```

Comment:
When the filter is done within the selector, it is memo'd and the component will only be updated if the end result (eg. after filtering) changes. With the change made here, it is going to re-render the component whenever any chat changes, and then re-run the filter each time.

It would be better if we could do this as part of the selector itself. I would probably refactor `useVenueChats` to take some options as props. In this case, I would probably call it something like `includeDeleted` or `withDeleted`, and default it to `false`. Then, based on this prop, we can do the basic selector, or the one that filters deleted.

To get the best memoisation, we want the selector function reference to remain the same, that is generally why we extract them to the selectors file. In this case it may make sense to do it more 'inline' within the hook though, so we could use `useCallback` to ensure our selector function reference only changes when the props it relies on changes. Something like the following (within the `useVenueChats` hook):

```typescript
const venueChatsSelector = useCallback((state) => {
  const venueChats = state.firestore.ordered.venueChats ?? [];

  if (withDeleted) {
    return venueChats;
  } else {
    return venueChats?.filter((chat) => chat.deleted !== true)
  }
}, [withDeleted])
```

If you wanted to keep the selector logic in the `selectors.ts` file, then we could basically extract the logic of the function described above into that file, and then just call that function within the `useCallback` within `useVenueChats` (which is probably the better way, as then all our selector logic is kept 'isolated' in that one file)

`selectors.ts`:
```typescript
const makeVenueChatsSelector = ({ withDeleted = false}) => (state) => {
  const venueChats = state.firestore.ordered.venueChats ?? [];

  if (withDeleted) {
    return venueChats;
  } else {
    return venueChats?.filter((chat) => chat.deleted !== true)
  }
}
```

`useVenueChats.ts`:
```typescript
const venueChatsSelector = useCallback((state) => makeVenueChatsSelector({ withDeleted })(state), [withDeleted])
```

Reply:
And if the above isn't done (which is my preferred choice), at the very very least we should be wrapping `filteredChats` with `useMemo` so it's not re-filtering on every render.

Reply:
I think this approach is what I'm striving to do as well. Did useMemo for now. Though, I don't think there is any memorization happening as we don't have libs like `reselect` for that. As far as I understand, `useSelect` serves only as a pure function to retrieve data from redux store. I can be mistaken though.

Reply:
useSelector definitely memo’s in 2 different ways without reselect:

- https://react-redux.js.org/api/hooks

Reply:
Backlogged in https://github.com/sparkletown/internal-sparkle-issues/issues/111